src/widget/widget.cpp

```diff
@@ -498,6 +498,19 @@ void Widget::onTransferClicked()
     activeChatroomWidget = nullptr;
 }
 
+bool Widget::confirmExecutableOpen(const QFileInfo file)
+{
+    if (file.isExecutable())
```

Comment:
You can't use QFileInfo::isExecutable on windows and more generally on filesystems that don't have permissions, because it always returns true.
And on Linux/Mac file transfer doesn't preserve permissions, so there's no way to detect if a received file is executable, it would always be false (since default perms are non-executable).

Maybe check if the filename finishes with common executable/dangerous extensions.
Here's a list from google : .exe, .com, .pif, .msi, .msp, .src, .app, .dmg, .hta, .cpl, .msc, .jar, .py, .bat, .vb, .vbs, .vbe, .js, .jse, .ws, .wsf, .wsc, .wsh, .ps1, .ps1xml, .ps2, .ps2xml, .psc1, .psc2, .msh, .msh1, .msh2, .mshxml, .msh1xml, .msh2xml, .scf, .lnk, .reg